test/built-ins/Set/prototype/union/array-throws.js

```diff
@@ -0,0 +1,17 @@
+// Copyright (C) 2023 Anthony Frehner. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+esid: pending
+description: Set.prototype.union doesn't work with arrays
```

Comment:
Unless... Consider adding a test where you add `size`, `has` and `keys` methods to an array or to `Array.prototype`. That would catch someone incorrectly special-casing arrays.

Reply:
Good idea, added set-like-array.js with this test.